src/nfts/NftGallery.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { RefreshControl, ScrollView, StyleSheet, Text, View } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { useDispatch, useSelector } from 'react-redux'
+import Touchable from 'src/components/Touchable'
+import ImageErrorIcon from 'src/icons/ImageErrorIcon'
+import DrawerTopBar from 'src/navigator/DrawerTopBar'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import NftImage from 'src/nfts/NftImage'
+import NftsLoadError from 'src/nfts/NftsLoadError'
+import { myNftsErrorSelector, myNftsLoadingSelector, myNftsSelector } from 'src/nfts/selectors'
+import { fetchMyNfts } from 'src/nfts/slice'
+import { Nft, NftOrigin } from 'src/nfts/types'
+import colors from 'src/styles/colors'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+
+// This is used instead of a gap between the images because the gap is not supported in RN yet.
+const imageSize = Math.min(variables.width / 2.5, variables.height / 4)
+
+function NftTouchable({ nft }: { nft: Nft }) {
+  return (
+    <Touchable
+      key={`${nft.contractAddress}-${nft.tokenId}`}
+      onPress={() => navigate(Screens.NftsInfoCarousel, { nfts: [nft] })}
+      style={styles.touchableIcon}
+    >
+      <NftImage
+        nft={nft}
+        testID="NftGallery/NftImage"
+        width={imageSize}
+        height={imageSize}
+        ErrorComponent={
+          <View style={styles.errorView}>
+            <ImageErrorIcon color="#C93717" />
+          </View>
+        }
+        origin={NftOrigin.NftGallery}
+        borderRadius={Spacing.Smallest8}
+      />
+    </Touchable>
+  )
+}
+
+export default function NftGallery() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const loading = useSelector(myNftsLoadingSelector)
+  const error = useSelector(myNftsErrorSelector)
+  const nfts = useSelector(myNftsSelector)
+
+  function onRefresh() {
+    dispatch(fetchMyNfts())
+  }
+
+  // TODO: add analytics event for page load
+
+  if (error) {
```

Comment:
nit: if you move the error handling inside the main return block, you could save some code (don't have to declare the safe area / drawer top bar twice, you already check for error in the main block anyway)